lib/components/overlays/quick_settings/pages/qs_language_page.dart

```diff
@@ -42,24 +42,39 @@ class QsLanguagePage extends StatelessWidget {
       body: Padding(
         padding: const EdgeInsets.symmetric(vertical: 16.0),
         child: ListView.builder(
-          itemCount: Locales.supported.length,
+          itemCount: locales.supportedLocales.length,
           itemBuilder: (context, index) {
             return ListTile(
               shape: RoundedRectangleBorder(
                 borderRadius: BorderRadius.circular(8),
               ),
-              leading: Flag.fromString(
-                Locales.supported[index].languageCode.replaceAll("en", "us"),
-                width: 32,
-                replacement: const Icon(Icons.language_outlined),
+              leading: Text(
+                locales.supportedLocales[index]
+                    .toString()
+                    .substring(3)
+                    .replaceAllMapped(
+                      RegExp('[A-Z]'),
+                      (match) => String.fromCharCode(
+                        match.group(0)!.codeUnitAt(0) + 127397,
+                      ),
+                    ),
+                style: const TextStyle(
+                  fontSize: 30,
+                ),
               ),
               title: Text(
-                localeNativeNames[Locales.supported[index].languageCode] ??
+                localeNativeNames[
+                        locales.supportedLocales[index].languageCode] ??
                     "Language code not found",
               ),
-              subtitle: Text(Locales.supported[index].languageCode),
+              subtitle: Text(locales.supportedLocales[index].toLanguageTag()),
+              trailing: Text(
+                "${locales.progressData[locales.supportedLocales[index].toLanguageTag()]} / ${locales.progressData[context.fallbackLocale.toLanguageTag()]}",
+              ),
               onTap: () {
-                context.setLocale(Locales.supported[index]);
+                context.locale =
+                    locales.supportedLocales[index].toFlutterLocale();
```

Comment:
To avoid this flutter locale conversion you can get them already converted using context.supportedLocales